test/test_pipeline.py

```diff
@@ -496,7 +512,7 @@ def test_array(self, filename):
             arrays = list(r.iterator(chunk_size=100))
             assert len(arrays) == 11
             concat_array = np.concatenate(arrays)
-            np.testing.assert_array_equal(array, concat_array)
+            assert compare_structured_arrays(np.concatenate(arrays), concat_array) == True
```

Comment:
when evaluating against boolean conditions, the standard is to use the `is` operator instead of `__eq__` to ensure you're not verifying the output of a `def __eq__(self, other)` but actually proving what you are checking in is in fact a boolean.

Reply:
81203f1f1ae7ffed41bcdcaf057c4897f1a83b38